synfig-studio/src/synfigapp/actions/valuedescset.cpp

```diff
@@ -209,75 +209,77 @@ Action::ValueDescSet::prepare()
 					ValueBase svalue(0);
 
 					if(index==origin_index) {
 						Point p = value.get(Point());
-						ValueBase parentBone_vb = (*bone->get_link("parent"))(time);
-						ValueBase parentBone_vb1 = (*bone1->get_link("parent"))(time);
+						ValueBase parentPoseBone_vb = (*pose_bone->get_link("parent"))(time);
+						ValueBase parentRestBone_vb = (*rest_bone->get_link("parent"))(time);
 
-						Bone parentBone;
-						Bone parentBone1;
+						Bone parentPoseBone;
+						Bone parentRestBone;
 
-						if (parentBone_vb.get_type() == type_bone_valuenode) {
-							parentBone = (*parentBone_vb.get(ValueNode_Bone::Handle()))(time).get(Bone());
-						} else if (parentBone_vb.get_type() == type_bone_object) {
-							parentBone = parentBone_vb.get(Bone());
+						if (parentPoseBone_vb.get_type() == type_bone_valuenode) {
+							parentPoseBone = (*parentPoseBone_vb.get(ValueNode_Bone::Handle()))(time).get(Bone());
+						} else if (parentPoseBone_vb.get_type() == type_bone_object) {
+							parentPoseBone = parentPoseBone_vb.get(Bone());
 						} else {
 							throw Error(_("Parent bone type not supported"));
 						}
-						if (parentBone_vb1.get_type() == type_bone_valuenode) {
-							parentBone1 = (*parentBone_vb1.get(ValueNode_Bone::Handle()))(time).get(Bone());
-						} else if (parentBone_vb1.get_type() == type_bone_object) {
-							parentBone1 = parentBone_vb1.get(Bone());
+						if (parentRestBone_vb.get_type() == type_bone_valuenode) {
+							parentRestBone = (*parentRestBone_vb.get(ValueNode_Bone::Handle()))(time).get(Bone());
+						} else if (parentRestBone_vb.get_type() == type_bone_object) {
+							parentRestBone = parentRestBone_vb.get(Bone());
 						} else {
 							throw Error(_("Parent bone type not supported"));
 						}
 
-						Matrix parentAnimMatrix = parentBone.get_animated_matrix();
-						Matrix parentAnimMatrix1 = parentBone1.get_animated_matrix();
+						const Matrix parentPoseAnimMatrix = parentPoseBone.get_animated_matrix();
+						const Matrix parentRestAnimMatrix = parentRestBone.get_animated_matrix();
 
-						p += parentAnimMatrix.get_transformed(bone->get_link(index)->operator()(time).get(Point()));
-						p -= parentAnimMatrix1.get_transformed(-bone1->get_link(index)->operator()(time).get(Point()));
-						svalue = ValueBase(parentAnimMatrix.get_inverted().get_transformed(p));
+						p += parentPoseAnimMatrix.get_transformed(pose_bone->get_link(index)->operator()(time).get(Point()));
+						p -= parentRestAnimMatrix.get_transformed(-rest_bone->get_link(index)->operator()(time).get(Point()));
+						svalue = ValueBase(parentPoseAnimMatrix.get_inverted().get_transformed(p));
 					}else if(index==angle_index){
 						Angle a = value.get(Angle());
-						a+=bone->get_link(index)->operator()(time).get(Angle());
-						a-=bone1->get_link(index)->operator()(time).get(Angle());
+						a += pose_bone->get_link(index)->operator()(time).get(Angle());
+						a -= rest_bone->get_link(index)->operator()(time).get(Angle());
```

Comment:
I suggest to switch order of those two lines - it will be easier to understand what's going on ^__^